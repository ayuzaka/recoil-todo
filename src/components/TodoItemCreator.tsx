import { VFC, useState, ChangeEvent } from 'react'
import { useTodoList } from '../hooks/useTodoList'
import { Input } from './Atoms/Input'
import { Button } from './Atoms/Button'

type Props = {
  inputValue: string
  onChange: (event: ChangeEvent<HTMLInputElement>) => void
  onClick: () => void
}

const Component: VFC<Props> = ({ inputValue, onChange, onClick }) => (
  <div>
    <Input type="text" value={inputValue} onChange={onChange} />
    <Button onClick={onClick}>Add</Button>
  </div>
)

const Container: VFC = () => {
  const [inputValue, setInputValue] = useState('')
  const { addItem } = useTodoList()

  const onClick = () => {
    addItem(inputValue)
    setInputValue('')
  }

  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value)
  }

  return <Component {...{ inputValue, onChange, onClick }} />
}

export const TodoItemCreator = Container
