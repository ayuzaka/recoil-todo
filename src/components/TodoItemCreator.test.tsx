import { RecoilRoot } from 'recoil'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { TodoItemCreator } from './TodoItemCreator'

describe('初期状態', () => {
  test('render', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <TodoItemCreator />
      </RecoilRoot>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('テキストボックスには何も入力されていないこと', () => {
    render(
      <RecoilRoot>
        <TodoItemCreator />
      </RecoilRoot>
    )
    expect(screen.getByRole('textbox')).toHaveValue('')
  })
})

describe('イベント', () => {
  beforeEach(() => {
    render(
      <RecoilRoot>
        <TodoItemCreator />
      </RecoilRoot>
    )
  })

  test('テキストボックスに入力できる', () => {
    userEvent.type(screen.getByRole('textbox'), 'Hello')
    expect(screen.getByRole('textbox')).toHaveValue('Hello')
  })

  test('ボタンを押したとき、テキストボックスの値がリセットされること', () => {
    const textbox = screen.getByRole('textbox')
    userEvent.type(textbox, 'Hello')
    expect(textbox).toHaveValue('Hello')
    userEvent.click(screen.getByRole('button'))
    expect(textbox).toHaveValue('')
  })

  test.todo('ボタンを押したとき、アイテムを追加する関数が実行されること')
})
