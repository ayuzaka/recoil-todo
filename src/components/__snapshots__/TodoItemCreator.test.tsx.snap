// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期状態 render 1`] = `
<DocumentFragment>
  <div>
    <input
      class="css-154kdnc"
      type="text"
      value=""
    />
    <button
      class="css-104jfzb"
      type="button"
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;
